### PR TITLE
Make all tests functions and structs crate-private

### PR DESCRIPTION
**Pull request by 0xPoe**
part of https://github.com/rust-lang/rustup/issues/2730

### CODE REVIEW

**Comment by 0xPoe** [comment]
I found a lot of methods and variables that are not being used, should I remove them all?

cc: @kinnison @rbtcollins

**Comment by 0xPoe** [comment]
/hold

Waiting for anyhow wroks laned.

**Comment by 0xPoe** [comment]
/unhold

I think it's ready for review.

**Comment by kinnison** [comment]
I'm not entirely sure if this is valuable here.  The way our tests are structured makes it quite hard to know for sure about which bits are used in which test files.  Also `pub` of any kind inside `test-$foo.rs` seems a little odd to me anyway.

I'm going to defer to @rbtcollins for this particular PR.

**Comment by 0xPoe** [comment]
Because our test is a standalone crate, it warns on cargo check, but not on cargo clippy.

I don't have too good a solution for it.

**Comment by kinnison** [comment]
I wonder if, instead, what we need is to restructure our tests/ dir into tests/dist.rs and tests/cli.rs with clitools/ and clitests/ as subdirs - i.e. build the dist tests (which import the rustup module) and the cli tests as two integration binaries rather than the nearly 10 we have now.  Then the clitools stuff becomes a module used by only one binary (the cli tests) and as such controlling unused functions etc. becomes much more plausible.  If we did that though, we'd need to be sure that we'd bottomed the potentially dodgy parallelism in some of the tests.

File: tests/mock/clitools.rs [hunk 1]
```diff
@@ -28,31 +28,31 @@ use crate::mock::topical_doc_data;
 use crate::mock::{MockComponentBuilder, MockFile, MockInstallerBuilder};
 
 /// The configuration used by the tests in this module
-pub struct Config {
+pub(crate) struct Config {
     /// Where we put the rustup / rustc / cargo bins
-    pub exedir: PathBuf,
+    pub(crate) exedir: PathBuf,
```
Comment (rbtcollins) [inline tests/mock/clitools.rs:33]: we shouldn't do this - just the pub(crate) on the struct, per our discussions and that bug.

Comment (0xPoe) [reply tests/mock/clitools.rs:33]: No, if we remove the modifier, it will be private. The bug fixed is the use of pub in pub(crate).

Comment (0xPoe) [reply tests/mock/clitools.rs:33]: Could this be another bug?

Comment (rbtcollins) [reply tests/mock/clitools.rs:33]: I'm not suggesting private, I'm suggesting simple `pub`

Comment (0xPoe) [reply tests/mock/clitools.rs:33]: Oh, you mean in this case we don't need to change it to pub(crate) but keep it pub?

**Comment by rbtcollins** [comment]
Exactly

**Comment by 0xPoe** [comment]
Close it for now, as it causes cargo check to misrepresent a lot of unused warnings. It can be very disruptive until we can sort out and re-architect the tests.

**Pull request closed without merging**